src/Microsoft.Health.Fhir.Shared.Core/Features/Search/Expressions/Parsers/ExpressionParser.cs

```diff
@@ -61,50 +59,72 @@ public Expression Parse(string resourceType, string key, string value)
             EnsureArg.IsNotNullOrWhiteSpace(key, nameof(key));
             EnsureArg.IsNotNullOrWhiteSpace(value, nameof(value));
 
-            // Split the key by the chain separator.
-            PathSegment[] paths = key.Split(new[] { SearchParams.SEARCH_CHAINSEPARATOR }, StringSplitOptions.RemoveEmptyEntries)
-                 .Select(path => PathSegment.Parse(path))
-                 .ToArray();
+            return ParseImpl(resourceType, key.AsSpan(), value);
+        }
 
-            if (paths?.Length == 0)
+        private Expression ParseImpl(string resourceType, ReadOnlySpan<char> key, string value)
+        {
+            if (TryConsume("_has:".AsSpan(), ref key))
```

Comment:
could `"_has:".AsSpan()` be stored as constant?

Reply:
It doesn't look like you can have a ReadOnlySpan<char> as a const, but I did make the string `_has:` a const.